llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -1735,7 +1736,8 @@ Instruction *InstCombinerImpl::foldBinopOfSextBoolToSelect(BinaryOperator &BO) {
   Constant *Zero = ConstantInt::getNullValue(BO.getType());
   Value *TVal = Builder.CreateBinOp(BO.getOpcode(), Ones, C);
   Value *FVal = Builder.CreateBinOp(BO.getOpcode(), Zero, C);
-  return SelectInst::Create(X, TVal, FVal);
+  return createSelectInstMaybeWithUnknownBranchWeights(X, TVal, FVal,
+                                                       BO.getFunction());
```

Comment:
```suggestion
  return createSelectInstWithUnknownBranchWeights(X, TVal, FVal);
```
This is a bit of a mouthful, I don't think the "maybe" adds anything here.

InstCombine always works on a single function, so there really shouldn't be a need to explicitly pass it anywhere. If InstCombiner doesn't store the Function yet, it's fine to add a member for it.

Reply:
Done (but I decided to just call this `createSelectInst(...)` as the function is usable when unknown branch weights are not appropriate).